inspirehep/modules/workflows/tasks/manual_merging.py

```diff
@@ -110,16 +110,29 @@ def save_roots(obj, eng):
     """
     def _merge_roots(new_uuid, head_roots, update_roots):
         """Return the new roots to link to head."""
-        head_sources = [h.source for h in head_roots]
+        head_sources = {h.source: h for h in head_roots}
         for u in update_roots:
-            if u.source not in head_sources:
+            if u.source not in head_sources.keys():
                 head_roots.append(
                     WorkflowsRecordSources(
                         source=u.source,
                         record_id=new_uuid,
                         json=u.json
                     )
                 )
+            else:
+                if head_sources[u.source].updated < u.updated:
+                    head_roots.remove(head_sources[u.source])
+                    db.session.delete(head_sources[u.source])
+                    head_roots.append(
+                        WorkflowsRecordSources(
+                            source=u.source,
+                            record_id=new_uuid,
+                            json=u.json
+                        )
+                    )
+            db.session.delete(u)
```

Comment:
why we have to delete the updated here?

Reply:
@drjova because we don't want to keep the roots of the record that got merged.